### PR TITLE
Change default OTLP port number

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This implements specification change https://github.com/open-telemetry/opentelemetry-specification/pull/1221

To make transition to new port numbers less painful OTLP receiver will
also accept data on the legacy port numbers when it is configured to
use the default endpoint. Users who use the default Collector config
can continue sending data to the legacy ports and have a graceful period
to update their senders to start sending to the new ports.

Note that OTLP/HTTP continues using a separate port number from OTLP/gRPC.
There is separate work in progress to use one port for both.

This must be merged after https://github.com/open-telemetry/opentelemetry-specification/pull/1221 is merged.

### CODE REVIEW

**Review by bogdandrutu — COMMENTED** [review]
Consider to remove the build.log

**Comment by tigrannajaryan** [comment]
> Consider to remove the build.log

Removed.

File: receiver/otlpreceiver/otlp.go
Comment (bogdandrutu) [inline receiver/otlpreceiver/otlp.go:130]: can we move all this code to a "startInternal" instead of having everything in this anonymous func?

Comment (tigrannajaryan) [reply receiver/otlpreceiver/otlp.go:173]: Done.

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-approvers please review.

File: receiver/otlpreceiver/README.md
Comment (bogdandrutu) [inline receiver/otlpreceiver/README.md:62]: We should not have a default yet for this correct? Until you merge then I will push my PR to share same port

File: receiver/otlpreceiver/factory.go
Comment (bogdandrutu) [inline receiver/otlpreceiver/factory.go:43]: ditto

**Review by bogdandrutu — COMMENTED** [review]
Everything looks good except the http default port

File: receiver/otlpreceiver/README.md
Comment (tigrannajaryan) [reply receiver/otlpreceiver/README.md:62]: OK, I'll revert http port changes, but will keep gRPC port changes.

Comment (tigrannajaryan) [reply receiver/otlpreceiver/README.md]: Done.

File: receiver/otlpreceiver/factory.go
Comment (tigrannajaryan) [reply receiver/otlpreceiver/factory.go]: Removed.

**Comment by tigrannajaryan** [comment]
@bogdandrutu PTAL.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
